### DISCUSSION TITLE
don't adjust mania scroll speed to bpm changes within a map

### DISCUSSION THREAD

**Discussion by LastExceed** [ppy/osu#13159] [Ideas]
Even when using fixed scroll speed, BPM changes within the map still affect the scroll speed (relative to the map's default BPM). Most of the time mappers counter this using SVs to bring the scroll speed back to its default value, as drastic BPM changes are otherwise unplayable. This in turn makes SV mapping harder as things like teleport's now need to be normalized to a value that isn't x1.0, and throws off beginner mappers who don't know about this. I understand the appeal of BPM scaling from a game design POV, but it's pretty much incompatible with the user's ability to configure scroll speed as most players set their scroll speed as fast as they can read to maximize precision of visual aiming and minimize the number of notes that are on screen at a time (see also: lane covers).

Therefore I suggest making this effect exclusive to the "scale osu!mania scroll speed with BPM" option (if we're even going to bring that back in lazer, see #2870).

This would of course break every map that was re-normalized using SVs (which includes ranked maps), so we'd have to account for that when parsing legacy maps. There's also aspire-like maps that use BPM changes to work around the SV bounds, but that's a different topic. more to that here #10266

**Comment by peppy** [comment]
I don't *think* this is something we want because it basically limits how a mapper can... map.

**Comment by ReiFan49** [comment]
as an unranked mod sure it's a good idea to practice, but still a "constant speed mod" usually breaks what mapper's message throughout the beatmap itself

**Comment by LastExceed** [comment]
> it basically limits how a mapper can... map

@peppy please elaborate. How does this proposal limit anything ?

> a "constant speed mod" usually breaks what mapper's message

@ReiFan49 I am not proposing a constant speed mod (that would be #9802). This proposal only removes BPM based speed changes which can be replaced by SVs if necessary. And regarding "breaking" anything, I explained in the OP that this change needs to be accounted for when parsing legacy maps to avoid any existing maps being changed.

**Comment by abraker95** [comment]
To paraphrase @LastExceed, if you have a beatmap going at 180 BPM and want a short section that is 90 BPM you need:
```
uninherited timing point 90 BPM
inherited timing point 2.0x
uninherited timing point 180 BPM
inherited timing point 0.5x
```
Because uninherited timing points change scroll speed. So inherited timing points are needed to counteract that. This asks for uninherited timing points not to affect scroll speed so that those extra inherited timing points would not be needed. So only this will be needed:
```
uninherited timing point 90 BPM
uninherited timing point 180 BPM
```

**Comment by LastExceed** [comment]
almost. SV after 180 would be 1.0x , not 0.5x , and since every bpm change automatically resets SV to 1.0 you could leave out the last one, but that's beside the point. Point is, having to manually calculate and add SVs that cancel out the effects of BPM scaling is very annoying, especially for maps with accelerations/decelerations:
![](https://i.imgur.com/QtXS58h.png)
(this is from [a ranked map](https://osu.ppy.sh/beatmapsets/426638))

**Comment by peppy** [comment]
In osu! bpm and slider velocity is hard coupled. If you add a toggle of the effect of one of these two, mappers no longer know how to map.

If we are to make such a change it needs sufficient documentation and consideration for all rulesets, not just mania.

**Comment by LastExceed** [comment]
> bpm and slider velocity is hard coupled

you mean from an implementation side of things ? An implementation detail shouldn't matter to game design (unless it's so fundamental that it would be too much work to change now?)

> mappers no longer know how to map

I don't understand this part. Sure, if you haven't read the patch notes and are used to re-normalizing scroll speed using SVs then you might be confused for a moment, but mapping isn't witchcraft. Especially when we implement this as an explicit toggle as you suggested this really shouldn't be a problem.

> it needs sufficient documentation

"`BPM changes` set the grid on which you can place the notes. `SV changes` set the scroll speed relative to the user configured default speed." 

Honestly don't see much to document here.

> consideration for all rulesets

I'm not familiar enough with the other game modes to judge this

**Comment by peppy** [comment]
Closing this conversation for now since everything needs to be said has in my last comment. It can only change if due consideration is given and I cannot explain it any more clearly than I did.

We may reconsider this in a distant future, but it's not something that will happen overnight.
